### DISCUSSION TITLE
BitBadges Integration

### DISCUSSION THREAD

**Discussion by trevormil** [deso-protocol/dips#118] [Large CIPs]
The last CIP regarding BitBadges didn't gain much traction: https://github.com/bitclout/cips/discussions/105

So, I wanted to resurrect it here in a new one. I would like to integrate BitBadges into bitclout.com's frontend and potentially into the main chain itself.

BitBadges is a solution where user's can issue NFT badges to any other users. Think of them like boy scout or military badges. You can issue badges to any other user and they will live permanently on the blockchain. So once you earn a badge, no one can take it away from you because it is linked with your public key. You then start to build a collection of badges for your BitClout profile and you can display all your best badges somewhere on your profile page. This will 1) aid people in investment decisions as it assigns a better reputation score than just creator coin price which can be manipulated and 2) offer a crypto software solution to many of BitClout's existings problems like scammers, verification, etc and 3) offer a new cool way of interaction between creator and audience.

What problems does this solve for BitClout currently?

Verification - Accounts like @VerifiedProfile can issue verification badges and then all you need to do to check if someone is verified is check if they have the badge issued from them. If in the future, you don't like @VerfiedProfile anymore, you can just choose to recognize someone else as the official verifier. Different users can even choose to recognize different official verifiers. There doesn't have to be one unique one. Much better solution than the current solutions. 

Scammers - Same thing as with verification for scammers. Right now, if someone is new to BitClout there is really no way to tell if someone has a history of scams. With a scammer badge, new people can directly see who is a scammer by a scammer badge. 

Along with these two main problems, there are infinitely many use cases for BitBadges because each badge is unique. So many interesting use cases will pop up.

Examples: 
- Premium Content Access Tickets or Susbcriptions
- Recognition Badges
- Digital Collectibles for Events (i.e. go to CloutCon, get a collectible CloutCon badge)
- Many more

Implementation:

We are fully open source. We offer a public API that gives access to all badge data to whoever wishes. So if we ever act untruthfully in the future and you don't trust BitBadges anymore, you can just fork us and create a new BitBadges. 

To issue a badge, all you need to provide is jwt verification with BitClout Identity. Then, you can issue a badge to anyone.

The only time we charge money is for over 25 recipients on a single badge which is just to reduce API spam limit which we don't plan to offer that feature on bitclout.com. We are already implemented on CloutFeed and we would like to do a similar integration on bitclout.com as well.

We would love to discuss if you have any ideas for more transparency.

There is nothing to lose with integrating BitBadges. It is a cool feature that solves many of BitClout's problems and there will be absolutely no creator coins/money involved.

Would love to answer any questions.

See https://www.bitbadges.github.io for more details as well.

**Comment by tijno** [comment]
I think this would be a great idea especially if it can adopt some of the other proposals here #59 at the same time.

Implementing it on chain would also remove the need for IPFS/ETH dependencies i think right?

Aspects to consider:

* burn badges - eg if i get spammed by a ton of badges i dont care about, I should have the ability to burn them.

* badges may imply trust - but as we have seen by some fake / scam accounts that still got the verifiedprofile "badge" - this can cause problems. How to resolve this? Give badges to badges maybe ;) (jk). Im not sure there is a solution, but its worth keeping in mind. Maybe someone else has ideas.

* scam badge brings with it all sorts of issues and in itself could be a new attack vector. So for these badges to be assigned to a profile it should be done very carefully and should have a transparent way of retracting or challenging. Might be worth discussing this as part of the #108

**Reply by trevormil** [reply]
Yep, definitely.

Yes, implementing it on chain would remove the need for IPFS/ETH dependencies from a BitClout perspective, but we plan to do both implementations. Our long term goal is to be able to issue badges cross chain like from $ETH to $CLOUT wallet and other chains may not fully recognize the BitClout chain as trusted. So, we would also like to keep IPFS as a trusted file storage system for all chains.

Burn Badges - So badges are permanently on the blockchain once issued. But from a profile perspective like you see in CloutFeed, first you must accept/reject badges to be displayed on your profile. This means you will never have a badge displayed on your profile w/o your permission. So what about scammer badges? They just won't be displayed on profile but permanently on blockchain, so anyone can look them up. We also have many options to hide badges as well if you don't want it displayed anymore. Badges can't be "burned" but they can be marked as non visible which in essence is the same thing (like hiding posts on BitClout)

Badges may imply trust - We also have a collections feature which is non permanent. So VerifiedProfile creates a collection with all his verification badges. You can choose to trust the collection for all your verifications. This way they can add and remove verification badges to the collection as he pleases. He can take out scam accounts when he wishes to "unverify" them. If you don't like their verification system, there is no need to use that one. You can switch to someone else's verification collection. 

Scam Badges - Yea, you are right. This does need to be carefully implemented. But with the collections I mentioned above, this is totally possible. Just trust the scammer collection. If the "scammer" wins a challenge and gets it retracted, the badge just gets removed from the collection.

Collections open up a world of possibilities and remove that permanence factor as well. Also, collections will be much more efficient because it won't carry all the badge data as overhead. Just badge hashes, recipients of collection, and issuers of collection. 

If you are interested in a CloutFeed demo of collections, dm me on Discord and I can share.

**Comment by maebeam** [comment]
Thanks for the suggestion @trevormil. If you weren't aware, the solution the core devs would like to build is [covered in the FAQ](https://docs.bitclout.com/faq/bitclout-faq#what-parts-of-bitclout-are-centralized-at-bitclout-com-and-what-parts-are-decentralized-on-the-bitclout-blockchain-and-what-parts-of-the-bitclout-blockchain-are-still-semi-centralized).

> Verified checkmarks are not stored on the blockchain. This is for the same reason that removed profiles aren’t. Instead, the core dev team has plans to introduce an “association” primitive that will allow profiles to associate with one another, effectively creating a more decentralized version of the blue checkmark. For example, your profile can list associations with universities or workplaces that have been cryptographically signed.

I think this data is important enough to store on chain rather than a third party service.

**Reply by trevormil** [reply]
Ok. Have to say I am extremely disappointed. We have had a working NFT badge verification solution for two months now. First, we built an NFT solution on BitClout. Then, a month later you release NFTs. Next, we transition to focus on badge NFTs. Then, a month later you are releasing our exact project and taking credit for it. We have been trying endlessly on multiple occasions to reach out for integration and storing everything on chain but haven't got a response (See below ) We have been working tirelessly to get it integrated into apps like CloutFeed and we get treated like this. Extremely disappointed. No consideration at all to a project like ours. 

This isn't the first time this has happened. BitSwap, CloutPool, and many more projects have been fantastically built but died off because you didn't recognize them. There is so much politics in play with what third party apps are being approved and what isn't. You would never see Vitalik Buterin promote CryptoPunks and profit off them because that is not what a decentralized founder should be doing. It seems like the only reason you choose to promote certain third party apps is because you can make a quick buck on them. That is highly centralized to me.

Badges can still be used for so much more though and we would still like to see an integration happen if possible. Even if a verification solution is not a use case for BitBadges on BitClout, profile badges offer so many more interesting use cases that are not currently available. _**We have big plans for BitBadges in the crypto space and we hope to include BitClout in them.**_






Proof of Direct Contacts w/ Core Team:

Message to diamondhands on BitClout on 7/11/2021
"Hey would love to discuss @BitBadges and the NFT launch with you. I think they are mutually beneficial and will really change the way we use BitClout. Basically BitBadges offers an NFT solution for BitClout but the NFTs are non transferable hence badges. So once you earn an NFT badge it is on the blockchain and can never be taken away from your bitclout public key"


He then tells me that they have had an idea about "associations" in the past but to make a CIP to get ball rolling. So, he acknowledged they haven't even begun implementing this yet. Then, he goes ahead and implements the idea without even contacting or getting in touch once after this. He knew we were building this project and goes and builds it and takes credit for it. I thought BitClout was supposed to be community based. 


Then over the past month since 7/11, I have sent the following four messages with no response.


1) Yea the “associations” idea is exactly what we are building! Each association however is marked with an NFT badge (picture, title, description, url, etc). Our current setup is we mint on IPFS and post the hashes on the BitClout chain via a post. We are preparing to launch on CloutFeed within the next week, and I can give you a sneak peak before that if you are interested. 

2) Hey just launched on CloutFeed in its newest update with @BitBadges if you want to check it out! Pretty much the associations idea linked with NFTs! Would love to talk about a couple things regarding the project with you (getting it integrated into bitclout.com frontend and also seeing if there is a better method to storing on chain)

3) Question: If I do the work in getting BitBadges integrated into bitclout.com frontend and create a PR, is that something that would interest you? I would like to get approval before I start so I don’t waste my time.

4) Would love to discuss on these topics. Made two CIPs to get the ball rolling:
https://github.com/bitclout/cips/discussions/117
https://github.com/bitclout/cips/discussions/118
The first one (#117) is about the core team's decision making and my complaints with it.

The second one is about integrating bitbadges into bitclout. I think it solves a lot of problems that you are currently having


Other Examples of Times I have Reached Out:

https://github.com/bitclout/cips/discussions/118
https://github.com/bitclout/cips/discussions/105

https://bitclout.com/posts/e0dad64f12965243f7b07530229a8250aba96144d226500b2053d852144c411f?feedTab=Following&tab=posts
https://bitclout.com/posts/cf09e8b408f9dc2147dad875ef01cdeb7eb81623e9405702179065d6579bf87f?feedTab=Following&tab=posts
https://bitclout.com/posts/8e0cf4debfa9710ac280a0a86b87c7b7a8cac8f3767f9d7549e9580bdb1dab4d?feedTab=Following&tab=posts
https://bitclout.com/posts/342c79866465d5d5a64e5e2ecd629758ff4333bb166aaa18be004576e31df2ec?feedTab=Following&tab=posts
https://bitclout.com/posts/3624c44777af634cb07056e87a2e67afa54074f103fe343e6a21cd7dadf348de?feedTab=Following&tab=posts
https://bitclout.com/posts/46befb950d1a8438838876b560198182c57efedeaf26ef49ad56d2f90fb5984d?feedTab=Following&tab=posts

Message to maebeam on discord on 7/3/2021:
trevormil — 07/03/2021
Hey! Wanted to ask a quick question regarding NFTs on BitClout. I am the cofounder of BitBadges which is an NFT solution on BitClout but they are non transferable (hence badges). To clarify, are all NFTs going to be transferable? Or are there going to be badge like ones which would render our project useless?
If not would love to talk about a potential roadmap of what we plan with BitBadges and how they could help BitClout become huge

**Reply by trevormil** [reply]
I have proof that diamondhands had acknowledged they were not working on the "associations" idea over a month ago and that we said we had a working implementation. 

This isn't the first time this has happened. The same thing happened w/ CloutPool when you promoted CloutReactor and called them the first mining solution. Over $80k in clout creator coin were dumped from CloutPool because of this. Is this what decentralized founders should be doing? Picking and choosing winners? Crypto is about decentralization and no barriers to entry. BitClout s looking more like a ponzi scheme right now than a true crypto decentralized protocol.

**Reply by trevormil** [reply]
Like I said, we have big plans for BitBadges in the crypto space. Would love to discuss everything we have in mind with you. We hope we can include BitClout in our plans as well, but we will move on if necessary.

**Reply by maebeam** [reply]
To clarify, no work has been done on associations. I just wanted to make sure you were aware of the associations idea and that it could be used to create badges.

**Reply by trevormil** [reply]
Apologies. From your comment, it made it seem like this was the solution you already had implemented or were currently implementing. I take back a lot of what I said and sorry for jumping to conclusions. I still feel that the communication was poor, but I was mistaken on the other points. 

Yes, I was made aware of the "associations" idea by diamondhands in DMs. I told him that was exactly what we are building with BitBadges and sent 4 more messages to him regarding it, sent multiple posts, and even made a CIP, and no response or communication. Especially with developers, I feel there needs to be a lot better communication. Also, I feel there needs to be a better job done with not picking and promoting certain community projects because that is just stifling competition. The core team should remain quiet on this and let both competition and the community decide what product is best.

As for your other points, I agree. It is valuable enough to store on chain which is why I have been reaching out for help/approval to get it integrated on chain. We already have a fully working implementation of the associations idea and would love to work with the core team to get it implemented. That is our goal and why I have been reaching out. Even if we don't use BitBadges for verification, I would love to include badges into the bitclout.com frontend because it is a great new feature that has many exciting use cases that weren't previously possible.

As for more of the implementation details, it is already artificially "on chain" using the IPFS and posting the hashes to @BitBadgesHash account. This could definitely be optimized though with an actual integration. Our vision w/ BitBadges is to be able to issue these badges cross chain (for example, from an $CLOUT wallet to $ETH wallet), so we would like to keep our IPFS/other decentralized storage systems in addition to a main chain integration. This is because some $ETH wallets may not trust the $CLOUT chain and thus, they can trust IPFS instead. With this cross chain issuing, we can help BitClout become much bigger in the whole crypto landscape. With badges for all your wallets, BitClout becomes your social identity for all your crypto wallets. This could be absolutely huge if implemented correctly.

We want to build the "associations" idea for you. We have a working implementation already. Would you be willing to let us?

**Comment by theparkerazzi** [comment]
Good Morning Trevor. My nodes are pretty much kind of up. I definitely want to continue the implementation of your project.
